### ISSUE TITLE
Error after after hot update: "The schema has already been initialized"

### ISSUE THREAD

**Issue by kigorw** [aws-amplify/amplify-js#5101]
Nextjs is doing hot module update when you change the code and you get error. 

Problem is here https://github.com/aws-amplify/amplify-js/blob/master/packages/datastore/src/datastore/datastore.ts#L105

Maybe would be better to just return current schema. Also it's probably not the best design decision to execute code during import phase because of unexpected side effects.

**Comment by glauberfc** [comment]
+1

**Comment by tombuckley91** [comment]
Subscribed, thank you @ericclemmons for your work on this, it's so frustrating when working in Next.

**Comment by mdegrees** [comment]
Here is a hack that worked for me: 

instead of this: 
`` import { Post } from "src/models" ``
Do this: 
```javascript
import { initSchema } from "@aws-amplify/datastore";
import { schema } from "src/models/schema";

let models;
if (typeof window !== "undefined") {
  models = initSchema(schema);
}
```

**Comment by ericclemmons** [comment]
This has been fixed as part of #6146.  See this main thread for more info: https://github.com/aws-amplify/amplify-js/issues/5435#issuecomment-692990722
